### PR TITLE
fix unpublished cycle

### LINKED ISSUE #1504: Publicly accessing non published cycle error

### PR DESCRIPTION
**Pull request by rfontanarosa**
close #1504

### CODE REVIEW

File: src/client/pages/Assessment/Assessment.tsx [hunk 1]
```diff
@@ -69,6 +72,8 @@ const Assessment: React.FC = () => {
 
   if (!assessment) return null
 
+  if (!Authorizer.canView({ countryIso, assessment, cycle, user })) window.location.href = BasePaths.Root()
```
Comment (sorja) [inline src/client/pages/Assessment/Assessment.tsx:75]: `history.replace` ?

Comment (minotogna) [reply src/client/pages/Assessment/Assessment.tsx:75]: actually yes, but get the history from react-router

Comment (rfontanarosa) [reply src/client/pages/Assessment/Assessment.tsx:75]: it's complicated. we can'y use useHistory at the same level of <Routes> definition (it is null).

Comment (minotogna) [reply src/client/pages/Assessment/Assessment.tsx:75]: useNavigate ? https://reactrouter.com/en/main/hooks/use-navigate

Comment (sorja) [reply src/client/pages/Assessment/Assessment.tsx:75]: just as info,
using `window.location.href` creates a new ajax query and leaves the SPA
while using the react-router stays inside SPA and does not create unnecessary ajax calls

Comment (minotogna) [reply src/client/pages/Assessment/Assessment.tsx:75]: yes, that's why I'm suggesting that

**Review by sorja — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**